### PR TITLE
Document all remaining undocumented symbols added since 1.1.1

### LINKED ISSUE #13192: Undocumented symbols added since 1.1.1
We have added ~23~ ~13~ ~5~  public functions and ~12~ ~3~ 4 macros since the release of 1.1.1 that have not been documented.

These should all be documented before beta1, or removed from the public API.

(NOTE the ~strike~ items have been addressed). 
````
$ perl util/find-doc-nits -o -e
````
crypto: function ASN1_item_sign_ex(3) undocumented
crypto: function ASN1_item_verify_ctx(3) undocumented
crypto: function ASN1_item_verify_ex(3) undocumented
~crypto: function ERR_load_CMP_strings(3) undocumented~
~crypto: function ERR_load_CRMF_strings(3) undocumented~
~crypto: function ERR_load_ESS_strings(3) undocumented~
~crypto: function ERR_load_HTTP_strings(3) undocumented~
~crypto: function ERR_load_KDF_strings(3) undocumented~
~crypto: function ERR_load_OSSL_DECODER_strings(3) undocumented~
~crypto: function ERR_load_OSSL_ENCODER_strings(3) undocumented~
crypto: function OPENSSL_LH_flush(3) undocumented
crypto: function asn1_d2i_read_bio(3) undocumented
~crypto: function evp_pkey_get_EC_KEY_curve_nid(3) undocumented~
~crypto: function EVP_PKCS82PKEY_ex(3) undocumented~
~crypto: function EVP_hex2ctrl(3) undocumented~
~crypto: function EVP_str2ctrl(3) undocumented~
~crypto: function EVP_aes_128_siv(3) undocumented~
~crypto: function EVP_aes_192_siv(3) undocumented~
~crypto: function EVP_aes_256_siv(3) undocumented~
~crypto: function ossl_b2i(3) undocumented~
~crypto: function ossl_b2i_bio(3) undocumented~
~crypto: function ossl_do_PVK_header(3) undocumented~
~crypto: function ossl_do_blob_header(3) undocumented~

# ~23~ ~16~ 5 libcrypto names are not documented
include/openssl/core_dispatch.h: macro OSSL_CORE_MAKE_FUNC(3) undocumented
include/openssl/opensslv.h: macro OPENSSL_VERSION_PREREQ(3) undocumented
include/openssl/params.h: macro OSSL_PARAM_DEFN(3) undocumented
include/openssl/x509v3.h: macro EXT_UTF8STRING(3) undocumented
~include/openssl/macros.h: macro OSSL_DEPRECATED(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATED_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_3_0_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_1_1_1_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_1_1_0_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_1_0_2_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_1_0_1_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_1_0_0_FOR(3) undocumented~
~include/openssl/macros.h: macro OSSL_DEPRECATEDIN_0_9_8_FOR(3) undocumented~
# ~12~ ~3~ 4 macros undocumented (count is approximate)

### PR DESCRIPTION
**Pull request by mattcaswell**
All remaining symbols from #13192 are documented.

Fixes #13192.

### CODE REVIEW

File: doc/man3/OPENSSL_LH_COMPFUNC.pod [hunk 1]
```diff
@@ -162,34 +179,47 @@ that is provided by the caller):
...
 
 =head1 NOTE
 
 The LHASH code is not thread safe. All updating operations, as well as
-B<lh_I<TYPE>_error>() call must be performed under a write lock. All retrieve
-operations should be performed under a read lock, I<unless> accurate
-usage statistics are desired. In which case, a write lock should be used
-for retrieve operations as well. For output of the usage statistics,
-using the functions from L<OPENSSL_LH_stats(3)>, a read lock suffices.
+B<lh_I<TYPE>_error>() or OPENSSL_LH_error() callS must be performed under
```
Comment (t8m) [inline doc/man3/OPENSSL_LH_COMPFUNC.pod:217]: s/callS/calls/

File: doc/man3/OPENSSL_LH_COMPFUNC.pod [hunk 2]
```diff
@@ -162,34 +179,47 @@ that is provided by the caller):
...
+NULL is returned if there is no such value in the hash table.
 
-B<lh_I<TYPE>_retrieve>() returns the hash table entry if it has been found,
-NULL otherwise.
+B<lh_I<TYPE>_retrieve>() and OPENSSL_LH_retrieve() return the hash table entry
+if it has been found, NULL otherwise.
 
-B<lh_I<TYPE>_error>() returns 1 if an error occurred in the last operation, 0
-otherwise. It's meaningful only after non-retrieve operations.
+B<lh_I<TYPE>_error>() OPENSSL_LH_error() return 1 if an error occurred in the
```
Comment (t8m) [inline doc/man3/OPENSSL_LH_COMPFUNC.pod:207]: missing and

File: doc/man3/OPENSSL_LH_COMPFUNC.pod [hunk 3]
```diff
@@ -162,34 +179,47 @@ that is provided by the caller):
...
 
-B<lh_I<TYPE>_new>() returns NULL on error, otherwise a pointer to the new
-B<LHASH> structure.
+B<lh_I<TYPE>_new>() and OPENSSL_LH_new() return NULL on error, otherwise a
+pointer to the new B<LHASH> structure.
 
-When a hash table entry is replaced, B<lh_I<TYPE>_insert>() returns the value
-being replaced. NULL is returned on normal operation and on error.
+When a hash table entry is replaced, B<lh_I<TYPE>_insert>() or
+OPENSSL_LH_insert() returns the value being replaced. NULL is returned on normal
```
Comment (t8m) [inline doc/man3/OPENSSL_LH_COMPFUNC.pod:198]: should it be returns or return?

**Review by t8m — COMMENTED** [review]
Just a few nits

**Comment by mattcaswell** [comment]
All nits fixed in the latest fixup. Please take another look.

**Review by t8m — APPROVED** [review]
LGTM

File: doc/man3/OPENSSL_LH_COMPFUNC.pod [hunk 4]
```diff
@@ -162,34 +179,48 @@ that is provided by the caller):
 B<lh_I<TYPE>_error>() can be used to determine if an error occurred in the last
 operation.
 
+OPENSSL_LH_new() is the same as the lh_TYPE_new() except that it is not type
+specific. So instead of returning an LHASH_OF(TYPE) value it is simply a void *.
```
Comment (paulidale) [inline doc/man3/OPENSSL_LH_COMPFUNC.pod:183]: s/it is simply/it returns/ ?

File: doc/man3/OpenSSL_version.pod [hunk 5]
```diff
@@ -73,6 +75,12 @@ B<OPENSSL_VERSION_TEXT> is a convenience macro to get a full descriptive
 version text, which includes B<OPENSSL_FULL_VERSION_STR> and the release
 date.
 
+B<OPENSSL_VERSION_PREREQ> is a useful macro for checking whether the OpenSSL
+version for the headers in use are at least at the given pre-requistive major
```
Comment (paulidale) [inline doc/man3/OpenSSL_version.pod:79]: s/are/is/
s/requistive/requisite/

**Review by paulidale — APPROVED** [review]
A couple of minor wordings but good to go once they've been looked at.

**Comment by mattcaswell** [comment]
Fixups pushed addressing feedback from @paulidale.

@t8m - Please can you reconfirm?

**Review by t8m — APPROVED** [review]

File: doc/man3/OPENSSL_LH_COMPFUNC.pod [hunk 6]
```diff
@@ -162,34 +179,48 @@ that is provided by the caller):
...
 operation.
 
+OPENSSL_LH_new() is the same as the lh_TYPE_new() except that it is not type
+specific. So instead of returning an LHASH_OF(TYPE) value it returns a void *.
+In the same way the functions OPENSSL_LH_free, OPENSSL_LH_flush,
+OPENSSL_LH_insert, OPENSSL_LH_delete, OPENSSL_LH_retrieve, OPENSSL_LH_doall,
+OPENSSL_LH_doall_arg, and OPENSSL_LH_error are equivalent to the similarly named
+lh_TYPE functions except that they return or use a void * where the equivalent
+lh_TYPE function returns or uses a TYPE * or LHASH_OF(TYPE) *. lh_TYPE functions
+are impleented as type checked wrappers around the OPENSSL_LH functions. Most
```
Comment (slontis) [inline doc/man3/OPENSSL_LH_COMPFUNC.pod:189]: implemented

File: doc/man3/OSSL_CORE_MAKE_FUNC.pod [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ #define OSSL_CORE_MAKE_FUNC(type,name,args)
+
+=head1 DESCRIPTION
+
+There are certain macros that may appear in OpenSSL header files that are
+reserved for internal use. They should not be used by applications or assumed
+to exist.
+
+All the macros listed in the synopsis above are reserved.
+
```
Comment (slontis) [inline doc/man3/OSSL_CORE_MAKE_FUNC.pod:20]: Added in section?

Comment (mattcaswell) [reply doc/man3/OSSL_CORE_MAKE_FUNC.pod:20]: I don't understand what change you are suggesting here?

Comment (slontis) [reply doc/man3/OSSL_CORE_MAKE_FUNC.pod:20]: Do you need a HISTORY section?

Comment (mattcaswell) [reply doc/man3/OSSL_CORE_MAKE_FUNC.pod:20]: Fixed.

Comment (mattcaswell) [reply doc/man3/OSSL_CORE_MAKE_FUNC.pod:20]: Fixed.

**Review by slontis — APPROVED** [review]
LGTM..

**Comment by slontis** [comment]
Maybe close and open this PR again.. The fix has been merged.

**Comment by slontis** [comment]
If it was all green before the NIT changes should be ok.

**Review by paulidale — APPROVED** [review]
still good

File: doc/man3/OPENSSL_LH_COMPFUNC.pod [hunk 8]
```diff
@@ -162,34 +179,48 @@ that is provided by the caller):
...
 
+OPENSSL_LH_new() is the same as the lh_TYPE_new() except that it is not type
+specific. So instead of returning an LHASH_OF(TYPE) value it returns a void *.
+In the same way the functions OPENSSL_LH_free, OPENSSL_LH_flush,
+OPENSSL_LH_insert, OPENSSL_LH_delete, OPENSSL_LH_retrieve, OPENSSL_LH_doall,
+OPENSSL_LH_doall_arg, and OPENSSL_LH_error are equivalent to the similarly named
+lh_TYPE functions except that they return or use a void * where the equivalent
+lh_TYPE function returns or uses a TYPE * or LHASH_OF(TYPE) *. lh_TYPE functions
+are implemented as type checked wrappers around the OPENSSL_LH functions. Most
+applications should not call the OPENSSL_LH functions directly.
```
Comment (levitte) [inline doc/man3/OPENSSL_LH_COMPFUNC.pod:190]: man-pages(7) doesn't say anything specific about how types should be formatted.  Looking through diverse man pages, I see them sometimes in bold, and sometimes in italics.  In our pages, we tend to use bold for the verbatim types.  What we've also done lately, when there's that metaphorical "TYPE", is to make it bold italics (i.e. italics markup inside the bold markup).

So in this paragraph, I can see the following markup additions:

`void *` => `B<void *>`
`LHASH_OF(TYPE)` => `B<LHASH_OF(I<TYPE>)>`
`LHASH_OF(TYPE) *` => `B<LHASH_OF(I<TYPE>) *>`
`TYPE *` => `B<I<TYPE> *>`

Functions need some markup too.  If we have `()` at the end of the function name, perldoc does that for us.  I see that you haven't done that for `OPENSSL_LH_free`, `OPENSSL_LH_flush`, ...
There are times when perldoc needs a bit of help, though, so please change:

`lh_TYPE_new()` => `B<lh_I<TYPE>_new>()`
`lh_TYPE functions` => `B<lh_I<TYPE>> functions`
`OPENSSL_LH functions` => `B<OPENSSL_LH> functions`

File: doc/man3/OSSL_PARAM_int.pod [hunk 9]
```diff
@@ -171,6 +174,12 @@ The storage for this parameter is at I<address> and is of I<size> bytes.
 OSSL_PARAM_END provides an end of parameter list marker.
 This should terminate all OSSL_PARAM arrays.
 
+The OSSL_PARAM_DEFN macro provides the ability to construct a single OSSL_PARAM
```
Comment (levitte) [inline doc/man3/OSSL_PARAM_int.pod:177]: `OSSL_PARAM_DEFN` => `OSSL_PARAM_DEFN()`
`OSSL_PARAM` => `B<OSSL_PARAM>`

**Review by levitte — CHANGES_REQUESTED** [review]

**Comment by mattcaswell** [comment]
New commit addressing latest feedback. Please reconfirm again everyone!

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed. Thanks!

**Pull request closed without merging**